src/object.rs

```diff
@@ -1051,3 +1060,78 @@ impl<T: IsA<Object> + StaticType + UnsafeFrom<ObjectRef> + Wrapper + ?Sized> Wea
     }
 }
 
+pub struct BindingBuilder<'a, S: IsA<Object> + 'a, T: IsA<Object> + 'a> {
+    source: &'a S,
+    source_property: &'a str,
+    target: &'a T,
+    target_property: &'a str,
+    flags: ::BindingFlags,
+    transform_to: Option<::Closure>,
+    transform_from: Option<::Closure>,
+}
+
+impl<'a, S: IsA<Object> + 'a, T: IsA<Object> + 'a> BindingBuilder<'a, S, T> {
+    fn new(source: &'a S, source_property: &'a str, target: &'a T, target_property: &'a str) -> Self {
+        Self { source, source_property, target, target_property, flags: ::BindingFlags::DEFAULT, transform_to: None, transform_from: None }
+    }
+
+    fn transform_closure<F: Fn(&::Binding, &Value) -> Option<Value> + Send + Sync + 'static>(func: F) -> ::Closure {
+        ::Closure::new(move |values| {
+            assert_eq!(values.len(), 3);
+            let binding = values[0].get::<::Binding>().unwrap();
+            let from = unsafe {
+                let ptr = gobject_ffi::g_value_get_boxed(mut_override(&values[1] as *const Value as *const gobject_ffi::GValue));
+                assert!(!ptr.is_null());
+                &*(ptr as *const gobject_ffi::GValue as *const Value)
+            };
+
+            match func(&binding, &from) {
+                None => Some(false.to_value()),
+                Some(value) => {
+                    unsafe {
+                        gobject_ffi::g_value_set_boxed(mut_override(&values[2] as *const Value as *const gobject_ffi::GValue), &value as *const Value as *const _);
+                    }
+
+                    Some(true.to_value())
+                }
+            }
+        })
+    }
+
+    pub fn transform_from<F: Fn(&::Binding, &Value) -> Option<Value> + Send + Sync + 'static>(self, func: F) -> Self {
+        Self {
+            transform_from: Some(Self::transform_closure(func)),
+            ..self
+        }
+    }
+
+    pub fn transform_to<F: Fn(&::Binding, &Value) -> Option<Value> + Send + Sync + 'static>(self, func: F) -> Self {
+        Self {
+            transform_to: Some(Self::transform_closure(func)),
+            ..self
+        }
+    }
+
+    pub fn flags(self, flags: ::BindingFlags) -> Self {
```

Comment:
Instead of this, one could have a `fn bidirectional(self) -> Self` etc for each of the flags